### PR TITLE
Fix search functionality in the new project layout

### PR DESCRIPTION
**Pull request by yuval-hazaz**
Issue Number: #3300 

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/CommandPalette/CommandPalette.tsx [hunk 1]
```diff
@@ -101,21 +102,27 @@ type Props = {
 };
 
 const CommandPalette = ({ trigger }: Props) => {
-  const match = useRouteMatch<{ resourceId: string }>("/:resourceId/");
+  const { currentWorkspace, currentProject, currentResource } = useContext(
+    AppContext
+  );
 
-  const { resourceId } = match?.params || {};
+  const projectBaseUrl = useMemo(
```
Comment (barshimi) [inline packages/amplication-client/src/CommandPalette/CommandPalette.tsx:109]: we need to add a generic function to create this url and check for undefined entities.

File: packages/amplication-client/src/CommandPalette/CommandPalette.tsx [hunk 2]
```diff
@@ -101,21 +102,27 @@ type Props = {
...
+  );
 
   const history = useHistory();
   const [query, setQuery] = useState("");
   const handleChange = (inputValue: string, userQuery: string) => {
     setQuery(userQuery);
   };
   const { data } = useQuery<TData>(SEARCH, {
-    variables: { query },
+    variables: { query, projectId: currentProject?.id },
```
Comment (barshimi) [inline packages/amplication-client/src/CommandPalette/CommandPalette.tsx:120]: do we need to add skip ?

Comment (yuval-hazaz) [reply packages/amplication-client/src/CommandPalette/CommandPalette.tsx:120]: no, it brings all data when it first opens

File: packages/amplication-client/src/CommandPalette/CommandPalette.tsx (hunk 1 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/CommandPalette/CommandPalette.tsx:109]: I agree, I event think it can be part of the context

**Pull request merged**